src/library/coap.cpp

```diff
@@ -860,16 +862,14 @@ void Coap::RequestsCache::Eliminate(const RequestHolder &aRequestHolder)
             break;
         }
     }
-}
-
-void Coap::RequestsCache::Clear()
-{
-    mRetransmissionTimer.Stop();
 
-    mContainer.clear();
+    if (mContainer.empty())
+    {
+        mRetransmissionTimer.Stop();
+    }
```

Comment:
```suggestion
    UpdateTimer()
```

We can simply put all timer starting and stopping logic to a single function (maybe called `UpdateTimer`) and call it whenever necessary.